pkg/cvo/sync_worker.go

```diff
@@ -751,12 +751,38 @@ func summarizeTaskGraphErrors(errs []error) error {
 	if len(errs) == 1 {
 		return errs[0]
 	}
+	// hide the generic "not available yet" when there are more specific errors present
+	if filtered := filterErrors(errs, isClusterOperatorNotAvailable); len(filtered) > 0 {
+		return newMultipleError(filtered)
+	}
+	// if we're only waiting for operators, condense the error down to a singleton
 	if err := newClusterOperatorsNotAvailable(errs); err != nil {
 		return err
 	}
 	return newMultipleError(errs)
 }
 
+// filterErrors returns only the errors in errs which are false for all fns.
+func filterErrors(errs []error, fns ...func(err error) bool) []error {
+	var filtered []error
+Next:
+	for _, err := range errs {
+		for _, fn := range fns {
```

Comment:
nit:
can we replace this `break JUMP` with something that's more easier to follow..

like
```
matches := false 
loop fns as fn
    if fn == true matches <= true
if !matches append
```

Reply:
Yeah

Reply:
Fixed